deprecation-notice.php

```diff
@@ -0,0 +1,47 @@
+<?php
+// Add an admin notice for outdated plugin version
+add_action('admin_notices', 'notify_plugin_update');
+function notify_plugin_update() {
+    $dismissed_timestamp = get_option('onesignal_deprecation_notice_dismissed_time');
+    $time_passed = !$dismissed_timestamp || (time() - $dismissed_timestamp) > (7 * DAY_IN_SECONDS);
+    $cutoff_date = strtotime('2025-12-31 23:59:59'); // Set cutoff date and time
+
+    // Show the notice only if it's before the cutoff date and the notice has not been recently dismissed
+    if (time() <= $cutoff_date && $time_passed) {
+        echo '
+        <div class="notice notice-warning is-dismissible" id="onesignal-update-notice">
+            <p>
+              <strong>
+                OneSignal Push Important Update:</strong>
+                All push notification configuration options are moving to OneSignal.com.
+                Please ensure your settings are updated by December 13, 2024 to avoid any service disruptions.
```

Comment:
How can they move settings to dashboard, before the cutoff date? – My understanding is current iteration of Dashboard page will not be shown until cutoff date, so they cannot move / update any Settings until then?

If we enable the new version of Dashboard before cutoff, what's the point of the deprecation banner? Because if they upgrade before then, they don't see this warning anyway.

Maybe a save or export should be introduced and we say:

> `"Please ensure you Export your current settings by..."`